cmd/provisioner-localpv/app/config.go

```diff
@@ -151,9 +152,12 @@ func (c *VolumeConfig) GetPath() (string, error) {
 	//}
 
 	pvRelPath := c.pvName
-	path := filepath.Join(basePath, pvRelPath)
+	//path := filepath.Join(basePath, pvRelPath)
 
-	return c.validatePath(path)
+	return mHostPath.NewBuilder().
+		WithPathJoin(basePath, pvRelPath).
+		WithCheckf(mHostPath.IsNonRoot(), "failed to get path: root directory is not allowed").
+		ValidateAndBuild()
```

Comment:
Thanks for writing this. :+1: :100: 

A caller can achieve the same in multiple ways (builder should expose multiple styles -- builder should not limit or impose restrictions by limiting to a particular style/rule). For example:
```go
hostpath.NewBuilder().
  WithBasePath(basepath).
  WithRelativePath(pvRelPath).
  WithCheckf(
    mHostPath.IsNonRoot(), "path should not be a root directory: %s/%s", basepath, pvRelPath
  ).
  Build()
```

- Note the use of `assertion` type message, that directly maps to `IsNotRoot` & does not try to add info about what Build is trying to achieve.
- Most possibly Build will return the error as follows:
```go
  return errors.New("failed to build hostpath: %+v", b.errs)
```